src/bundle/mutable.js

```diff
@@ -0,0 +1,143 @@
+// Adapted from SES/Caja
+// Copyright (C) 2011 Google Inc.
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/startSES.js
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/repairES5.js
+
+export default function makeRepairDataProperties() {
+  const {
+    defineProperties,
+    getOwnPropertyDescriptors,
+    hasOwnProperty,
+  } = Object;
+  const { ownKeys } = Reflect;
+
+  // Object.defineProperty is allowed to fail silently,
```

Comment:
This was based off a comment I found in `realms-shim/src/common`, but doesn't seem to be true in at least node v10

Reply:
Another obscure corner of JavaScript. The only case where this might happen is on browsers, where there's a distinction between the reified so-called "WindowProxy" object and the unreified so-called "Window" object. The issue is that a browser frame, when it is navigated from url1 to url2, gets the completely new root realm associated with url2, including global variable bindings, but keeps the same global "window" object (which is actually a WindowProxy) because that is the same object the parent can hold onto to designate this frame no matter what it is navigated to.

The way this is implemented is that every root realm that occupied the frame has its own hidden so-called "Window" object, whose properties are aliased to that root realm's global variables. The so-called "WindowProxy" object has a current hidden "Window" object, that it forwards all operations to. Done naively, this leads to a contradiction:

Say `wp` is such a WindowProxy object. While it is at url1, and therefore forwarding to hidden window hw1, someone does an 

```js
Object.defineProperty(wp, 'foo', {
  value: 88,
  writable: false,
  configurable: false
});
```

Let's say that this call reports success, however `defineProperty` is supposed to report success. Then, presumably, hw1 actually has a non-writable, non-configurable `foo` data property with value 88. By the object invariants, these conditions represent a commitment that this property on this object must always be a non-writable, non-configurable `foo` data property with value 88. Fine so far.

Now some navigates the frame to url2, associated with hidden window hw2 that has no `foo` property, and whose root realm therefore also has no `foo` global variable. What should

```js
Object.getOwnPropertyDescriptor(wp, 'foo')
```
return? What it actually does across all browsers, and as mandated by the html spec, is to forward that `getOwnPropertyDescriptor` request to its *current* hidden window object, hw2, which returns undefined. However, were `wp` to do so, after the previous `defineProperty` had indicated success, it would violate the previous commitment.

Therefore, the previous call to `defineProperty` must not report success. Heroic work by Mozilla on Firefox at first had that `defineProperty` call throw an error, which was the only way that `defineProperty` could report that the operation failed. This broke some old code in some old version of some library that did not actually care whether the `defineProperty` succeeded, failed, or did anything at all. But if the call to `defineProperty` threw, the library broke.

Therefore, for this one horrible special case, we allow `defineProperty` to report failure by returning false rather than throwing. SES does not care about breaking that old version of some obscure library. SES does care about the threat to integrity if code proceeds normally under the assumption that some operation succeeded when it didn't. Thus, SES should restore the less kludgy behavior of `defineProperty`, that could only report failure by throwing.

Reply:
For now should I continue to use `Object.defineProperties` as workround? or something else?

Reply:
Attn @ljharb Do you know the status of the possibility that the failure masking behavior of `Object.defineProperty` be extended to `Object.defineProperties`? In your opinion, should we switch to another technique to unmask the failure masking, to stay safe in the face of this possible change?

Reply:
Good. I agree with the conclusion. I was asking because I recall some tc39 talk of extending the failure masking behavior to `Object.defineProperties` for consistency. But I haven't heard much. If you haven't run across this, it is likely inactive. Whew. Thanks.